server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -3423,12 +3432,22 @@ else if (!template.isPublicTemplate() && caller.getType() != Account.ACCOUNT_TYP
             return uniqueTmplPair;
         }
         List<TemplateJoinVO> uniqueTmpls = uniqueTmplPair.first();
-        String[] tzIds = new String[uniqueTmpls.size()];
         int i = 0;
-        for (TemplateJoinVO v : uniqueTmpls) {
-            tzIds[i++] = v.getTempZonePair();
+        List<TemplateJoinVO> vrs = null;
+        if (showUnique) {
+            Long[] tzIds = new Long[uniqueTmpls.size()];
```

Comment:
@davidjumani nit - let's refactor and use a more modern form of writing code, the current code looks less like Java, more like C/C++; can be written as:
```
Long[] templateIds = uniqueTmpls.stream().map(template -> template.getId()).toArray(Long[]::new);
```